inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp

```diff
@@ -738,714 +738,366 @@ struct typed_resizeLinearU8C3C4 {
-        uchar* out_rows[2] = { out.OutLineB(0), out.OutLineB(1) };
+        GAPI_Assert(0 <= outCoord && outCoord <= outSz-1);
+        GAPI_Assert(0 <= index0 && index0 < index1 && index1 <= inSz);
 
-        int buf_width = out.length();
+        Unit unit;
 
-        const auto rowFunc = type_dispatch<nv12_to_rgb_supported_types>(out.meta().depth, cv_type_id{}, typed_nv12_to_rgb_row<isa_tag_t>{}, nullptr);
+        unit.index0 = checked_cast<index_type>(index0);
+        unit.index1 = checked_cast<index_type>(index1);
 
-        GAPI_DbgAssert(rowFunc);
+        unit.alpha0 = convert_cast<alpha_type>(alpha0);
+        unit.alpha1 = convert_cast<alpha_type>(alpha1);
 
-        rowFunc(y_rows, uv_row, out_rows, buf_width);
+        return unit;
     }
-};
-
-GAPI_FLUID_KERNEL(FI420toRGB, I420toRGB, false) {
-    static const int Window = 1;
-    static const int LPI = 2;
-    static const auto Kind = cv::GFluidKernel::Kind::YUV420toRGB;
-
-    static void run(const cv::gapi::fluid::View & in_y,
-                    const cv::gapi::fluid::View & in_u,
-                    const cv::gapi::fluid::View & in_v,
-                    cv::gapi::fluid::Buffer & out) {
-        GAPI_DbgAssert(is_cv_type_in_list<i420_to_rgb_supported_types>(out.meta().depth));
 
-        const uchar* u_row = in_u.InLineB(0);
-        const uchar* v_row = in_v.InLineB(0);
-        const uchar* y_rows[2] = { in_y.InLineB(0), in_y.InLineB(1) };
-        uchar* out_rows[2] = { out.OutLineB(0), out.OutLineB(1) };
+    int    inSz, outSz;
+    double ratio, inv_ratio;
 
-        int buf_width = out.length();
-        GAPI_DbgAssert(in_u.length() == in_v.length());
+    alpha_type  alpha;  // == inv_ratio, rounded
 
-        const auto rowFunc = type_dispatch<i420_to_rgb_supported_types>(out.meta().depth, cv_type_id{}, typed_i420_to_rgb_row<isa_tag_t>{}, nullptr);
+    void init(int _inSz, int _outSz) {
+        inSz  = _inSz;
+        outSz = _outSz;
 
-        GAPI_DbgAssert(rowFunc);
+        inv_ratio = invRatio(inSz, outSz);
+        ratio     = 1.0 / inv_ratio;
 
-        rowFunc(y_rows, u_row, v_row, out_rows, buf_width);
+        alpha = convert_cast<alpha_type>(inv_ratio);
     }
 };
 
-GAPI_FLUID_KERNEL(FSplit2, Split2, false) {
-    static const int LPI = 4;
-    static const int Window = 1;
-    static void run(const cv::gapi::fluid::View & in,
-                    cv::gapi::fluid::Buffer & out1,
-                    cv::gapi::fluid::Buffer & out2) {
-        GAPI_DbgAssert(2 == in.meta().chan);
-        GAPI_DbgAssert(1 == out1.meta().chan);
-        GAPI_DbgAssert(1 == out2.meta().chan);
-        GAPI_DbgAssert(in.meta().depth == out1.meta().depth);
-        GAPI_DbgAssert(in.meta().depth == out2.meta().depth);
-        GAPI_DbgAssert(is_cv_type_in_list<split_supported_types>(in.meta().depth));
+namespace areaDownscale32f {
+struct Mapper: public AreaDownMapper<float, int, float> {
+    Mapper(int _inSz, int _outSz) {
+        init(_inSz, _outSz);
```

Comment:
looks like we can avoid `init` and just call `AreaDownMapper` constuctor

Reply:
Done.